utils/windowfunction/frameboundrange.cpp

```diff
@@ -299,21 +299,26 @@ void FrameBoundExpressionRange<T>::validate()
             }
 
             case execplan::CalpontSystemCatalog::DECIMAL:
+            case execplan::CalpontSystemCatalog::UDECIMAL:
             {
-                if (this->fRow.getColumnWidth(this->fIndex[1]) < 16)
+                if (UNLIKELY(this->fRow.getColumnWidth(this->fIndex[1])
+                                < datatypes::MAXDECIMALWIDTH))
                 {
-                    int64_t tmp = this->fRow.getIntField(this->fIndex[1]);
+                    uint64_t tmp = this->fRow.getIntField(this->fIndex[1]);
```

Comment:
This is not correct for narrow decimals; the rhs returns an int64_t for an UDECIMAL which is not correct. I think we should keep the original two separate cases for DECIMAL and UDECIMAL.

Reply:
I didn't get the point to be honest.